pkg/vm/createEvm.go

```diff
@@ -5,43 +5,48 @@ package vm
 import (
 	"bytes"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"math/big"
 
-	"github.com/ava-labs/avalanche-cli/cmd/prompts"
 	"github.com/ava-labs/avalanche-cli/ux"
 	"github.com/ava-labs/avalanchego/utils/logging"
 	"github.com/ava-labs/subnet-evm/core"
 	"github.com/ava-labs/subnet-evm/params"
 )
 
-func CreateEvmGenesis(name string, log logging.Logger) ([]byte, error) {
+func CreateEvmGenesis(name string, log logging.Logger) ([]byte, string, error) {
 	ux.Logger.PrintToUser("creating subnet %s", name)
 
 	genesis := core.Genesis{}
 	conf := params.SubnetEVMDefaultChainConfig
 
-	chainId, err := getChainId()
-	if err != nil {
-		return []byte{}, err
-	}
-	conf.ChainID = chainId
-
-	*conf, err = getFeeConfig(*conf)
-	if err != nil {
-		return []byte{}, err
-	}
-
-	allocation, err := getAllocation()
-	if err != nil {
-		return []byte{}, err
+	stage := startStage
+
+	var chainId *big.Int
+	var tokenName string
+	var allocation core.GenesisAlloc
+	var err error
+
+	for stage != doneStage {
+		switch stage {
+		case startStage:
+			stage = descriptorStage
+		case descriptorStage:
```

Comment:
This looks like a state machine:
I think it's clearer if the next stage would not come back from the function (unless, the next stage is dependent on the function's result, which doesn't seem to be the case), but is clearly set after the function is evaluated.

This allows to keep the stage flow in one place, helping to understand it more clearly.

Reply:
you know, stages are just ints, so maybe we could control stage ordering with addition, hmmmm

Reply:
I implemented this new state incrementing/decrementing strategy